mock.go

```diff
@@ -0,0 +1,71 @@
+//nolint:unused
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/stretchr/testify/mock"
+)
+
+type Mock struct {
+	mock.Mock
+}
+
+var _ Client = &Mock{}
+
+func (m *Mock) Get(ctx context.Context, key string, data interface{}) error {
+	args := m.Called(ctx, key, data)
+	if args.Error(0) != nil {
+		return args.Error(0)
+	}
+	return nil
+}
+
+func (m *Mock) Set(ctx context.Context, key string, data interface{}, expiration time.Time) error {
+	args := m.Called(ctx, key, data, expiration)
+	if args.Error(0) != nil {
+		return args.Error(0)
+	}
+	return nil
+}
+
+func (m *Mock) Add(ctx context.Context, key string, data interface{}, expiration time.Time) error {
+	args := m.Called(ctx, key, data, expiration)
+	if args.Error(0) != nil {
+		return args.Error(0)
+	}
+	return nil
+}
+
+func (m *Mock) Delete(ctx context.Context, key string) error {
+	args := m.Called(ctx, key)
+	if args.Error(0) != nil {
+		return args.Error(0)
+	}
+	return nil
```

Comment:
This can be simplified to be just `return args.Error(0)`

Reply:
Understood.